### PR TITLE
[SYNTH-12983] Add testOverrides properties to env variables

### PR DESCRIPTION
**Pull request by AntoineDona**
### What and why?

We wanted to add an option for testOverrides properties to be configured through ENV variable, starting with `DATADOG_SYNTHETICS_OVERRIDE_`. This PR adds ENV variable for the following properties: `allowInsecureCertificates`, `body`, `bodyType`, `defaultStepTimeout`, `executionRule`, `followRedirects`, `pollingTimeout`, `resourceUrlSubstitutionRegexes`, `retry`, `startUrl`, `startUrlSubstitutionRegex` and `testTimeout`

### How?

- Added the variables listed above as ENV variables
- Created a utility function toNumber to convert environment variables to numbers, while handling edge cases (undefined values).

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/synthetics/README.md [hunk 1]
```diff
@@ -355,7 +355,7 @@ Your test files must be named with a `.synthetics.json` suffix.
 
 The `<TEST_PUBLIC_ID>` can be either the identifier of the test found in the URL of a test details page (for example, for `https://app.datadoghq.com/synthetics/details/abc-def-ghi`, it would be `abc-def-ghi`) or the full URL to the details page (for example, directly `https://app.datadoghq.com/synthetics/details/abc-def-ghi`).
 
-All options under the `testOverrides` key are optional and allow overriding of the test configuration as stored in Datadog.
+All options under the `testOverrides` key are optional and allow overriding of the test configuration as stored in Datadog. These options can also be set via environment variables starting with `DATADOG_SYNTHETICS_OVERRIDE_`.
```
Comment (jhgilbert) [inline src/commands/synthetics/README.md:358]:
```suggestion
All options under the `testOverrides` key are optional and allow overriding of the test configuration as stored in Datadog. These options can also be set with environment variables starting with `DATADOG_SYNTHETICS_OVERRIDE_`.
```

**Review by jhgilbert — APPROVED** [review]
Approved with a minor suggestion, thanks!

File: src/commands/synthetics/run-tests-command.ts [hunk 2]
```diff
@@ -244,11 +244,31 @@ export class RunTestsCommand extends Command {
...
+        defaultStepTimeout: toNumber(process.env.DATADOG_SYNTHETICS_OVERRIDE_DEFAULT_STEP_TIMEOUT),
+        executionRule: toExecutionRule(process.env.DATADOG_SYNTHETICS_OVERRIDE_EXECUTION_RULE),
+        followRedirects: toBoolean(process.env.DATADOG_SYNTHETICS_OVERRIDE_FOLLOW_REDIRECTS),
+        resourceUrlSubstitutionRegexes: process.env.DATADOG_SYNTHETICS_OVERRIDE_RESOURCE_URL_SUBSTITUTION_REGEXES?.split(
+          ';'
+        ),
+        retry: Object.keys(retryConfig).length > 0 ? retryConfig : undefined,
+        startUrl: process.env.DATADOG_SYNTHETICS_OVERRIDE_START_URL,
+        startUrlSubstitutionRegex: process.env.DATADOG_SYNTHETICS_OVERRIDE_START_URL_SUBSTITUTION_REGEX,
+        testTimeout: toNumber(process.env.DATADOG_SYNTHETICS_OVERRIDE_TEST_TIMEOUT),
```
Comment (teodor2312) [inline src/commands/synthetics/run-tests-command.ts:271]: (nit) Can you order all all of these alphabetically? (It's okay if you include it as part of the next PR to just not have to deal with commit history)

File: src/commands/synthetics/__tests__/cli.test.ts [hunk 3]
```diff
@@ -99,6 +127,28 @@ describe('run-test', () => {
...
+          interval: 42,
+        },
+      }
+      await command['resolveConfig']()
+
+      expect(command['config'].defaultTestOverrides.retry).toEqual({
+        count: 5,
+        interval: 42,
+      })
+    })
```
Comment (teodor2312) [inline src/commands/synthetics/__tests__/cli.test.ts:150]: (nit) Technically to have complete test coverage you should have a test case for `interval` as well :D and it would be easy to parameterise the test

However considering you're testing the code path, i can let is slide :D

**Review by teodor2312 — APPROVED** [review]
Some nits but it's good to merge 👍
